clap/parsers.zig

```diff
@@ -29,6 +30,24 @@ test "string" {
     try testing.expectEqualStrings("aa", try string("aa"));
 }
 
+/// Parse boolean value. False, off and 0 considered false, everything else is true.
+pub fn boolean(in: []const u8) error{ParameterTooLong}!bool {
+    // check if input is not too long
+    const MAX_LEN = 16;
+    if (in.len >= MAX_LEN) return error.ParameterTooLong;
+
+    // convert to lowercase
+    var lower: [MAX_LEN]u8 = undefined;
+    for (in, 0..) |c, i| {
+        lower[i] = std.ascii.toLower(c);
+    }
+
+    if (std.mem.eql(u8, "0", lower[0..in.len]) or
+        std.mem.eql(u8, "off", lower[0..in.len]) or
+        std.mem.eql(u8, "false", lower[0..in.len])) return false;
+    return true;
+}
+
```

Comment:
Hmm. Idk about this one. This seems quite opinionated and I can easily imagine anyone would expect different behaviour for `bool` values.

* Not wanting the default to `true` behaviour. If someone passes `of` that does not seem like a `true` to me.
* `yes/no` could be wanted
* Someone might not wanna have `0/1` as that seems quite "programmer" specific

People should use flags for boolean options in most cases. If they don't want a default, they can easily just use `parsers.enumeration(enum { on, off })` or whatever else they like.

Reply:
Thanks for the feedback, I will make those changes.